### PR TITLE
feat: score calculator

### LINKED ISSUE #16: [FE] Scoring
Increase scores when follow the right pattern.

### PR DESCRIPTION
**Pull request by ArcaneSand**
## What I did

formular score calculate

## Note

This world is pain

### CODE REVIEW

File: apps/server/src/Scoring.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+let correctSequence = ['A', 'A', 'B', 'C', 'D'];
```
Comment (Imeany12) [inline apps/server/src/Scoring.ts]: If we are not gonna use this as a default sequence, should we make this an empty arr?

File: apps/server/src/Scoring.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+let noteIndex = 0;
+let combo = 0;
+
+function setNoteSequence(seq: string[]) {
+  correctSequence = seq;
+  noteIndex = 0;
+  UpdateCombo(false);
+}
```
Comment (Imeany12) [inline apps/server/src/Scoring.ts]: I think we should add return infer type in this function as well to keep the code consistency

File: apps/server/src/Scoring.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  if (!isCorrectNote) {
+    combo = 0; // Reset combo if the answer is wrong
+  } else {
+    combo++; // Increment combo if the answer is correct
+  }
+}
+
+function MoveNoteIndex() {
+  noteIndex++;
+}
```
Comment (Imeany12) [inline apps/server/src/Scoring.ts]: same as above

File: apps/server/src/Scoring.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+function checkNoteSequence(input: string): {
+  result: boolean;
+  score: number;
+  message: string;
+  curIndex: number;
+} {
```
Comment (Imeany12) [inline apps/server/src/Scoring.ts:31]: same as above

**Review by Imeany12 — CHANGES_REQUESTED** [review]

**Comment by ArcaneSand** [comment]
push

**Review by Imeany12 — APPROVED** [review]
LGTM

**Pull request merged**